webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -60,47 +61,111 @@ public function createAction(Request $request, ?Problem $problem = null): Respon
+            ->getForm();
 
-        if ($form->isSubmitted() && $form->isValid()) {
+        $formUpload->handleRequest($request);
+        $formPaste->handleRequest($request);
+        if ($formUpload->isSubmitted() || $formPaste->isSubmitted()) {
             if ($contest === null) {
                 $this->addFlash('danger', 'No active contest');
             } elseif (!$this->dj->checkrole('jury') && !$contest->getFreezeData()->started()) {
                 $this->addFlash('danger', 'Contest has not yet started');
             } else {
-                /** @var Problem $problem */
-                $problem = $form->get('problem')->getData();
-                /** @var Language $language */
-                $language = $form->get('language')->getData();
-                /** @var UploadedFile[] $files */
-                $files      = $form->get('code')->getData();
-                if (!is_array($files)) {
-                    $files = [$files];
+                $problem = null;
+                $language = null;
+                $files = [];
+                $entryPoint = null;
+                $message = '';
+        
+                if ($formUpload->isSubmitted() && $formUpload->isValid()) {
+                    $problem = $formUpload->get('problem')->getData();
+                    $language = $formUpload->get('language')->getData();
+                    $files = $formUpload->get('code')->getData();
+                    if (!is_array($files)) {
+                        $files = [$files];
+                    }
+                    $entryPoint = $formUpload->get('entry_point')->getData() ?: null;
+                } elseif ($formPaste->isSubmitted() && $formPaste->isValid()) {
+                    $problem = $formPaste->get('problem')->getData();
+                    $language = $formPaste->get('language')->getData();
+                    $codeContent = $formPaste->get('code_content')->getData();
+        
+                    if ($codeContent == null || empty(trim($codeContent))) {
+                        $this->addFlash('danger', 'No code content provided.');
+                        return $this->redirectToRoute('team_index');
+                    }
+        
+                    $tempDir = sys_get_temp_dir();
+                    $tempFileName = sprintf(
+                        'submission_%s_%s_%s.%s',
+                        $user->getUsername(),
+                        $problem->getName(),
+                        date('Y-m-d_H-i-s'),
+                        $language->getExtensions()[0]
+                    );
+                    $tempFileName = preg_replace('/[^a-zA-Z0-9_.-]/', '_', $tempFileName);
+                    $tempFilePath = $tempDir . DIRECTORY_SEPARATOR . $tempFileName;
+                    file_put_contents($tempFilePath, $codeContent);
+        
+                    $uploadedFile = new UploadedFile(
+                        $tempFilePath,
+                        $tempFileName,
```

Comment:
I think we should use a simpler name here (and then of course use that as base for the entry point calculation). A nice simple naming scheme could be the `shortName` of the `contestProblem`.

Reply:
I still want the pasted submission to have a special identifier in the filename, to differentiate it from uploads. Is it possible to do this?

Reply:
What is the reason the way of submitting is important? You could test with adding `fileupload-` to the name? You would need to test yourself if this still works.

Reply:
Why is it "fileupload"? I thought it would be something like "Paste."

However, I now think using "shortName" directly is fine too.

Reply:
If you think the short name as given by the contest problem is ok, let's use that.

Reply:
Currently, the short name allows various characters. I will still use ```preg_replace('/[^a-zA-Z0-9_.-]/', '_', $tempFileName);``` to handle file naming using the short name.